test/personality/personality-test.js

```diff
@@ -0,0 +1,409 @@
+const assert = require('assert');
+const should = require('chai').should();
+const expect = require('chai').expect;
+const proxyquire =  require('proxyquire')
+
+let helperStub = {};
+let personality;
+
+function setHelperStub() {
+  helperStub.random = function () { return 0 };
+}
+
+function unsetHelperStub() {
+  helperStub.random = function () { return Math.random() };
+}
+
+describe('Personality', () => {
```

Comment:
Now that we've pulled our personalities into data files, I am not sure if we should be testing them as thoroughly (since testing data files seems like overkill to me?). Do you think it would make sense to do just one thorough pass-through with a more complicated personality (like "who's a good bot")? Or perhaps it makes sense to fake out our "personality" file(s)?

And I was also wondering if we could set random to return 0 or 0.999, to test that randomness is actually working as expected, instead of using the true Math.random

This all sounds like a lot of work though, maybe we can make a different story to handle all this later?

Reply:
Yeah, it probably makes more sense and is more maintainable to just have well-crafted test input for the personalities; we don't necessarily want to test our specific personalities but rather whether the `personality` module properly selects responses for given input.  I agree that a different story is probably the better spot for that work.

I can also change the `Math.random` calls in `personality-test.js` to better test the randomness of responses.